crates/valence_client/src/layer.rs

```diff
@@ -0,0 +1,76 @@
+use std::collections::HashSet;
+
+use bevy_app::prelude::*;
+use bevy_ecs::prelude::*;
+use valence_core::layer::LayerType;
+
+use crate::FlushPacketsSet;
+
+pub(super) fn build(app: &mut App) {
+    app.add_system(update_old_client_layer_set.in_set(FlushPacketsSet));
+}
+
+/// A component that represents the layers that an client is on.
+/// This is used to determine which entity to send to a client.
+///
+/// Usage: `ClientLayerSet::new(vec![1, 2])`
+///
+/// This will create a new `ClientLayerSet` with layers 1 and 2.
+/// The entity that are on layer 1 and 2 will be sent to the client.
+/// as well as the entity that don't have a `Layer` component.
+#[derive(Component, Default, Debug)]
+pub struct ClientLayerSet {
+    pub layers: HashSet<LayerType>,
+    old_layers: HashSet<LayerType>,
+}
+
+impl ClientLayerSet {
+    pub fn new(layers: Vec<LayerType>) -> Self {
```

Comment:
Should `layers` be of type `impl Iterator<Output = LayerType>` to avoid heap allocation?